### PR TITLE
Add augmentation to synthetic tutorial

### PR DESCRIPTION
**Pull request by henryre**
* Adds some optional noise dimensions, and TFs that just resample new noise
* Moves to Pandas storage type

**Test plan**
```bash
tox -e dev
source .env/bin/activate
python3 tutorials/synthetic/synthetic.py
```

### CODE REVIEW

**Comment by henryre** [comment]
Fixing Travis error and adding unit tests

**Comment by henryre** [comment]
Adding to tasks: we should make the synthetic data tests proper unit tests (i.e. no dependence on PandasLF/TFApplier, etc.)

File: tutorials/synthetic/synthetic.py [hunk 1]
```diff
@@ -19,32 +22,42 @@
 #   3. And finally train an `EndModel` using the generated training labels
 
 #%%
-# Generating a simple mixture of gaussians dataset of d-dim one-hot vectors
-# as a pandas DataFrame
+# Generating a simple mixture of gaussians dataset of d-dim vectors
+# as a list of SimpleNamespaces
```
Comment (ajratner) [inline tutorials/synthetic/synthetic.py]: Wait aren't they `DataFrames` now?

Comment (henryre) [reply tutorials/synthetic/synthetic.py]: Wow I played myself

**Pull request merged**